src/dash-table/conditional/index.ts

```diff
@@ -29,17 +34,31 @@ export interface IEditableElement {
 }
 
 export type ConditionalBasicFilter = INamedElement & ITypedElement;
-export type ConditionalDataCell = IConditionalElement & IIndexedRowElement & INamedElement & ITypedElement & IEditableElement;
+export type ConditionalDataCell = IActiveElement & IConditionalElement & IIndexedRowElement & INamedElement & ITypedElement & IEditableElement;
 export type ConditionalCell = INamedElement & ITypedElement;
 export type ConditionalHeader = IIndexedHeaderElement & INamedElement & ITypedElement;
 
 function ifAstFilter(ast: QuerySyntaxTree, datum: Datum) {
     return ast.isValid && ast.evaluate(datum);
 }
 
+export function ifColumnActive(condition: IActiveElement | undefined, active: boolean, selected: boolean) {
+    if (!condition) {
+        return true;
+    }
+
+    // Apply active if active, apply selected if selected AND not active (active and selected styling don't overlap)
```

Comment:
It is pretty common to want some commonality - for example selection gets a certain background color, active gets that same background color plus a border. How hard would it be to allow this? Like, first you look for `is_active: true` conditions, then any attribute that's not already set can be set by an `is_selected: true` condition? Not terrible if we can't make that work, but it does seem like it would be more intuitive that way.

Reply:
Reworked this more deeply. Instead of `is_active` and `is_selected` using a single `state:'active'|'selected'` nested prop for both. Makes it impossible to make invalid combinations like `active=True, selected=False`. https://github.com/plotly/dash-table/pull/729/commits/43c00fbe4bd71a45c3d24db8fde7cb2881968d17